applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,709 @@
+		err1 = nrf_cloud_agps_process(evt->data.msg.buf, evt->data.msg.len, NULL);
+		if (err1) {
+			LOG_INF("Unable to process A-GPS data, error: %d", err1);
+		}
+#endif /* CONFIG_SLM_AGPS */
+
+#if defined(CONFIG_SLM_PGPS)
+		int err2;
+
+		err2 = nrf_cloud_pgps_process(evt->data.msg.buf, evt->data.msg.len);
+		if (err2) {
+			LOG_ERR("Unable to process P-GPS data, error: %d", err2);
+		}
+#endif /* CONFIG_SLM_PGPS */
+
+#if defined(CONFIG_SLM_CELL_POS)
+		struct nrf_cloud_cell_pos_result result;
+		int err3;
+
+		err3 = nrf_cloud_cell_pos_process(evt->data.msg.buf, &result);
+		if (err3 == 0) {
+			sprintf(rsp_buf, "\r\n#XCELLPOS: %d,%lf,%lf,%d\r\n",
+				result.type, result.lat, result.lon, result.unc);
+			rsp_send(rsp_buf, strlen(rsp_buf));
+		} else if (err3 == 1) {
+			LOG_WRN("No position found");
+		} else {
+			LOG_ERR("Unable to process cell pos data, error: %d", err3);
+		}
+#endif /* CONFIG_SLM_CELL_POS */
+}
+
+static void cloud_event_handler(const struct cloud_backend *const backend,
+				const struct cloud_event *const evt, void *user_data)
+{
+	ARG_UNUSED(backend);
+	ARG_UNUSED(user_data);
+
+	switch (evt->type) {
+	case CLOUD_EVT_CONNECTING:
+		LOG_DBG("CLOUD_EVT_CONNECTING");
+		break;
+	case CLOUD_EVT_CONNECTED:
+		LOG_INF("CLOUD_EVT_CONNECTED");
+		break;
+	case CLOUD_EVT_READY:
+		LOG_INF("CLOUD_EVT_READY");
+		on_cloud_evt_ready();
+		break;
+	case CLOUD_EVT_DISCONNECTED:
+		LOG_INF("CLOUD_EVT_DISCONNECTED");
+		break;
+	case CLOUD_EVT_ERROR:
+		LOG_ERR("CLOUD_EVT_ERROR");
+		break;
+	case CLOUD_EVT_DATA_SENT:
+		LOG_DBG("CLOUD_EVT_DATA_SENT");
+		break;
+	case CLOUD_EVT_DATA_RECEIVED:
+		LOG_INF("CLOUD_EVT_DATA_RECEIVED");
+		on_cloud_evt_data_received(evt);
+		break;
+	case CLOUD_EVT_PAIR_REQUEST:
+		LOG_DBG("CLOUD_EVT_PAIR_REQUEST");
+		break;
+	case CLOUD_EVT_PAIR_DONE:
+		LOG_DBG("CLOUD_EVT_PAIR_DONE");
+		break;
+	case CLOUD_EVT_FOTA_DONE:
+		LOG_DBG("CLOUD_EVT_FOTA_DONE");
+		break;
+	case CLOUD_EVT_FOTA_ERROR:
+		LOG_ERR("CLOUD_EVT_FOTA_ERROR");
+		break;
+	default:
+		break;
+	}
+}
+#endif
+
+/**@brief handle AT#XGPS commands
+ *  AT#XGPS=<op>[,<interval>[,<timeout>]]
+ *  AT#XGPS? READ command not supported
+ *  AT#XGPS=?
+ */
+int handle_at_gps(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+	uint16_t interval;
+	uint32_t timeout = 0;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == GPS_START) {
+			err = at_params_unsigned_short_get(&at_param_list, 2, &interval);
+			if (err < 0) {
+				return err;
+			}
+			(void)at_params_unsigned_int_get(&at_param_list, 3, &timeout);
+			/* GNSS API spec check */
+			if (interval == 0 || interval == 1 ||
+			   (interval >= 10 && interval <= 65535)) {
+				err = nrf_modem_gnss_fix_interval_set(interval);
+				if (err) {
+					LOG_ERR("Failed to set fix interval, error: %d", err);
+					return err;
+				}
+				err = nrf_modem_gnss_start();
```

Comment:
No need to configure fix retry? The default value of 60 seconds will be used. At least without A-GPS, this is quite a short time.

Reply:
Since it's AT command, we have the flexibility to support all single, continuous and periodic modes

Reply:
Yes, but for single and periodic fix modes the fix retry time will currently always be 60 seconds. That may be too short for the first fix, at least without A-GPS. Would it be good to add a new parameter in the AT commands for setting the fix retry time?

Reply:
Added as an optional parameter:
AT#XGPS=<op>[,<interval>[,<retry_period>]]
AT#XAGPS=<op>[,<interval>[,<retry_period>]]
AT#XPGPS=<op>[,<interval>[,<retry_period>]]